sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -58,7 +58,9 @@ case class CreateViewCommand(
   override def output: Seq[Attribute] = Seq.empty[Attribute]
 
   require(tableDesc.tableType == CatalogTableType.VIEW)
-  require(tableDesc.viewText.isDefined)
+  if (!isTemporary) {
+    require(tableDesc.viewText.isDefined)
```

Comment:
we need a better error message here. i know it's not your code but might as well fix it while you are at it.

also fix the previous line.